### ISSUE TITLE
[CI] CertGenUtilsTests testIsAnyLocalAddress failing

### ISSUE THREAD

**Issue by romseygeek** [elastic/elasticsearch#86418]
This test will fail on JDK19 because it tries to mock java.net.InetAddress which is final.

**Build scan:**
https://gradle-enterprise.elastic.co/s/6djzxfdilixus/tests/:x-pack:plugin:security:cli:test/org.elasticsearch.xpack.security.cli.CertGenUtilsTests/testIsAnyLocalAddress

**Reproduction line:**
`./gradlew ':x-pack:plugin:security:cli:test' --tests "org.elasticsearch.xpack.security.cli.CertGenUtilsTests.testIsAnyLocalAddress" -Dtests.seed=30DF9C9C7C3A0131 -Dtests.locale=ru -Dtests.timezone=Europe/Nicosia -Druntime.java=19`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.cli.CertGenUtilsTests&tests.test=testIsAnyLocalAddress

**Failure excerpt:**
```
org.mockito.exceptions.base.MockitoException: 
Cannot mock/spy class java.net.InetAddress
Mockito cannot mock/spy because :
 - sealed class

  at __randomizedtesting.SeedInfo.seed([30DF9C9C7C3A0131:6B0686B3376A207A]:0)
  at org.elasticsearch.xpack.security.cli.CertGenUtilsTests.testIsAnyLocalAddress(CertGenUtilsTests.java:101)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:828)

```

**Comment by tvernum** [comment]
@slobodanadamovic do you think you could look at this one, since you've recently done a little bit of work on certificate generation?

**Comment by slobodanadamovic** [comment]
@tvernum Sure, I will have a look.

**Comment by dimitris-athanasiou** [comment]
Got a failure of this in `7.17` branch: https://gradle-enterprise.elastic.co/s/q7k6rfrsqeraq/tests/:x-pack:plugin:security:cli:test/org.elasticsearch.xpack.security.cli.CertGenUtilsTests/testIsAnyLocalAddress?top-execution=1

@slobodanadamovic Could you backport the fix in `7.17` branch as well please?

**Comment by slobodanadamovic** [comment]
> @slobodanadamovic Could you backport the fix in `7.17` branch as well please?

Yes, I can do that. Thanks for reporting.

**Comment by slobodanadamovic** [comment]
Backported to 7.17 -> https://github.com/elastic/elasticsearch/pull/89503
